### ISSUE TITLE
Finding and replacing a string with another. Template processor

### ISSUE THREAD

**Issue by mudassar192** [PHPOffice/PHPWord#631]
Hi , 
 Currently I am working with PHPWord , and I am trying to replace one string with another.Is there some way that I can do this. I have looked into the TemplateProcessor but am unable to find any solution for this. Can u kindly guide me in this regard.

**Comment by ozilion** [comment]
For templates only you can replace by variables. There is no search and replace feature for templates, yet. But maybe you can read content of docx, and make string replace then save you docx as new file.

**Comment by mudassar192** [comment]
Yes sure, I tried to follow this approach that u just mentioned. but What i could understand was there was a phpword object --> section --> abstract-container--> addText() function.  but I was unable to understand where the text was actually stored. 
Basically my question simply put is "How to convert the phpword (containing the read document) into string". or how to extract text from a document object. 
P.S : your help is appreciated.

**Comment by ngocketit** [comment]
@ozilion Could you please tell how can I define variables in a docx template? I'm using Office for Mac but couldn't find any ways of doing so. Thanks!

**Comment by ozilion** [comment]
@ngocketit just put your variables between ${ and } without spacesand in your code, use as

setValue($variable, $value)

**Comment by ngocketit** [comment]
@ozilion Thanks a lot! It's as that simple and I got it work now.
